perf/bloom_filter/bloom_filter.cc

```diff
@@ -0,0 +1,297 @@
+	std::vector<Tuple> insertTuples;
+};
+
+/** Tuple that will make select query look up all tree. */
+Tuple *nonexistenetTuple;
+/** Struct that holds:
+ *  1. inserted tuples;
+ *  2. tuples for select queries.
+ */
+TupleData *data;
+
+/** Args passed through lua code. */
+struct BenchArgs {
+	/** Name of benchmarked space. */
+	const char *space_name;
+	/** Space id. */
+	uint32_t space_id;
+	/** Tuple count. */
+	uint32_t tuple_count;
+
+	explicit BenchArgs(const char *args, const char *args_end) {
+		uint32_t arg_count = mp_decode_array(&args);
+		uint32_t name_len = 0;
+		const char *name = mp_decode_str(&args, &name_len);
+		space_name = strndup(name, name_len);
+		space_id = mp_decode_uint(&args);
+		tuple_count = mp_decode_uint(&args);
+	}
+
+	BenchArgs(const BenchArgs &) = delete;
+
+	BenchArgs(BenchArgs &&other) = delete;
+
+	~BenchArgs() { free((void *)space_name); }
+};
+
+/** Trivial wrapper for storing benchmark execution time. */
+struct BenchTimeResult {
+	double elapsed_time;
+};
+
+/** Represents results of benchmark. */
+struct BenchAlgorithmResult {
+	BenchAlgorithmResult(const char *name) : space_name(name) {}
+
+	/** Name of benchmarking space. */
+	const char *space_name;
+	/** Primary index bloom filter binary size. */
+	size_t bloom_bsize = 0;
+	struct BenchTimeResult insertTimeMs;
+	struct BenchTimeResult selectTimeMs;
+};
+
+int
+init(box_function_ctx_t *ctx, const char *args, const char *args_end)
+{
+	uint32_t arg_count = mp_decode_array(&args);
+	uint32_t name_len = 0;
+	const char *name = mp_decode_str(&args, &name_len);
+	const char *build_name = strndup(name, name_len);
+	uint32_t tuple_count = mp_decode_uint(&args);
+
+	(void)ctx;
+	(void)args;
+	(void)args_end;
+
+	std::cout << std::endl;
+	std::cout << "-----------------------------------------------------------------" << std::endl;
+	std::cout << "STARTED benchmarking." << std::endl;
+
+	memset(tuple_string, '.', STRING_FIELD_SIZE);
+	nonexistenetTuple = new Tuple(tuple_count + 1);
+	data = new TupleData(tuple_count);
+	resultsFile.open(resultsFileName, std::ios_base::in);
+	if (!resultsFile) {
+		std::cout << "Results file "
+			  << resultsFileName
+			  << " does not exist. Creating it." << std::endl;
+		resultsFile.open(resultsFileName, std::ios_base::app);
+		resultsFile << "builds:" << std::endl;
+	} else {
+		std::cout << "Results file "
+			  << resultsFileName
+			  << " does exist. Starting to write." << std::endl;
+		resultsFile.close();
+		resultsFile.open(resultsFileName, std::ios_base::app);
+	}
+	resultsFile << "    - build:" << std::endl;
+	resultsFile << "            build_name: " << build_name << std::endl;
+	resultsFile << "            tuple_count: " << tuple_count << std::endl;
+	resultsFile << "            spaces:" << std::endl;
+	return 0;
+}
+
+int
+stop(box_function_ctx_t *ctx, const char *args, const char *args_end)
+{
+	(void)ctx;
+	(void)args;
+	(void)args_end;
+
+	delete data;
+	delete nonexistenetTuple;
+	resultsFile.close();
```

Comment:
This function is called by the destructor of `std::ofstream`.

Reply:
Removed this line.